File: Solutions/0003_LongestSubstringWithoutRepeatingCharacters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if(s == "")
            return 0;

        int retVal = 1;
        int curr = 0;
        uint bin = 0;
        vector<char> seenChars;
        for(int i = 0; i < s.size(); i++) {
            curr = 1;
            seenChars = {s[i]};
            for(int j = i + 1; j < s.size(); j++) {
                // Checks if s[j] has already been seen
                if(find(seenChars.begin(), seenChars.end(), s[j]) != seenChars.end()) 
                    j = s.size();
                else { // If not adds it to the vector and increases the count
                    seenChars.push_back(s[j]);
                    curr++;
                }
            }
            if(curr > retVal)
                retVal = curr;
        }
        return retVal;
    }
};

File: Solutions/0003_LongestSubstringWithoutRepeatingCharacters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if s == "":
            return 0
        if len(s) == 1:
            return 1
        
        longest_subset = 1
        for i, _s1 in enumerate(s):
            charset = {_s1}
            for j, _s2 in enumerate(s[i+1:]):
                if _s2 not in charset:
                    charset.add(_s2)
                else:
                    i = j
                    break
            longest_subset = max(longest_subset, len(charset))

        return longest_subset